setup.py

```diff
@@ -266,6 +267,28 @@
 import setuptools.command.sdist
 from setuptools import Extension, find_packages, setup
 from setuptools.dist import Distribution
+
+
+cwd = os.path.dirname(os.path.abspath(__file__))
+
+# Add the current directory to the Python path so that we can import `tools`.
+# This is required when running this script with a PEP-517-enabled build backend.
+#
+# From the PEP-517 documentation: https://peps.python.org/pep-0517
+#
+# > When importing the module path, we do *not* look in the directory containing
+# > the source tree, unless that would be on `sys.path` anyway (e.g. because it
+# > is specified in `PYTHONPATH`).
+#
+sys.path.insert(0, cwd)  # this only affects the current process
+# Add the current directory to PYTHONPATH so that we can import `tools` in subprocesses
+os.environ["PYTHONPATH"] = os.pathsep.join(
+    [
+        cwd,
+        os.getenv("PYTHONPATH", ""),
+    ]
+).rstrip(os.pathsep)
```

Comment:
As a follow up it's probably worth restructuring the logic here so we don't have to do this lol.